scripts/collection_from_items.py

```diff
@@ -112,28 +118,51 @@ def main() -> None:
 
     s3_client = client("s3")
 
-    files_to_read = list_json_in_uri(uri, s3_client)
+    files_to_read = list_files_in_uri(uri, (".json", "_footprint.geojson"), s3_client)
 
     start_time = time_in_ms()
+    polygons = []
     for key, result in get_object_parallel_multithreading(
         bucket_name_from_path(uri), files_to_read, s3_client, arguments.concurrency
     ):
-        item_stac = json.loads(result["Body"].read().decode("utf-8"))
-
-        if not arguments.collection_id == item_stac.get("collection"):
-            get_log().trace(
-                "skipping: item.collection != collection.id",
-                file=key,
-                action="collection_from_items",
-                reason="skip",
-            )
-            continue
-
-        collection.add_item(item_stac)
-        get_log().info("item added to collection", item=item_stac["id"], file=key)
+        content = json.loads(result["Body"].read().decode("utf-8"))
+
+        if key.endswith(".json"):
+            if not arguments.collection_id == content.get("collection"):
+                get_log().trace(
+                    "skipping: item.collection != collection.id",
+                    file=key,
+                    action="collection_from_items",
+                    reason="skip",
+                )
+                continue
+            collection.add_item(content)
+            get_log().info("item added to collection", item=content["id"], file=key)
+        elif key.endswith("_footprint.geojson"):
+            get_log().debug(f"adding geometry from {key}")
+            geom = shapely.geometry.shape(content["features"][0]["geometry"])
+            polygons.append(geom)
```

Comment:
I can't quite tell whether there should be an `else` here. Shouldn't we throw in that case?

Reply:
We are listing only files that are `.json` or `_footprint.geojson`. If there was another kind of file that would mean `list_files_in_uri()` has a problem?

Reply:
OK, so this duplicates part of the functionality of `list_files_in_uri`. In that case, how much work would it be to refactor so that we don't have to check the suffix again in this function? For example, we could change `list_files_in_uri` to return a *tuple* of lists of strings corresponding to the files matching each of the `suffixes`, then handle those separately.

Reply:
As discussed, it won't be optimal to return different lists per suffix as we would need to call `get_object_parallel_multithreading` multiple time rather than once.